nengo_sphinx_theme/ext/resolvedefaults.py

```diff
@@ -10,59 +20,61 @@ class TestClass:
     int_param = IntParam("int_param", default=1)
     str_param = StringParam("str_param", default="hello")
 
-    def __init__(self, int_param=Default, str_param=Default):
-        pass
+    def __init__(self, int_param=Default, str_param=Default, module_param=np.random):
+        """Init method"""
+
+    def another_method(self, module_param=np.random):
+        """A method"""
 
 
 class DisplayDefault:
     def __init__(self, value):
         self.value = value
 
     def __repr__(self):
-        return "Default<{!r}>".format(self.value)
+        return self.value
 
 
-def resolve_default(cls, arg, value):
-    if value is not Default:
-        return value
+def resolve_default(cls, param):
+    if inspect.ismodule(param.default):
+        return DisplayDefault(param.default.__name__)
+    elif param.default is not Default:
+        return param.default
     else:
-        for param in (getattr(cls, name) for name in iter_params(cls)):
-            if param.name == arg:
-                return DisplayDefault(param.default)
+        for cls_param in (getattr(cls, name) for name in iter_params(cls)):
+            if cls_param.name == param.name:
+                return DisplayDefault("Default<{!r}>".format(cls_param.default))
         warnings.warn(
             "Default value for argument {} of {} could not be "
-            "resolved.".format(arg, cls)
+            "resolved.".format(param.name, cls)
         )
-        return value
+        return param.default
 
 
 def autodoc_defaults(app, what, name, obj, options, signature, return_annotation):
-    if what != "class":
+    if what not in ("function", "method", "class"):
         return None
-    spec = inspect.getfullargspec(obj.__init__)
 
-    if spec.defaults is None or not any(val is Default for val in spec.defaults):
+    spec = inspect.signature(obj)
```

Comment:
One difference I noticed is that class signatures no longer contain `self` at the beginning. Also, the signature for `SignalDict` is gone. However, when I go back and build with `master` in my current setup, these changes are also there, so it might be unrelated to this PR. However, `self` is [still appearing in the latest nengo-sphinx-theme docs](https://www.nengo.ai/nengo-sphinx-theme/extensions.html#nengo_sphinx_theme.ext.resolvedefaults.TestClass), so I'm not quite sure why I'm seeing a difference.

`SignalDict` is a little special because it's a subclass of `dict` and doesn't modify `__init__`. I was getting errors with it when building the Nengo docs with this code, because `inspect.signature` doesn't work for built-in things very well it seems. So I added some code to cover that (as well as any built-in functions we try to get signatures for, like `time.time` in `nengo.utils.stdlib`). Its signature disappearing also happens on `master` for me, though, so again not related to these changes I guess, but I don't know why.

Reply:
So `self` is also not appearing in `gh-pages-test`, even though it's in the lastest master docs in `gh-pages`. Also, re-doing things locally, `self` is now appearing in `master` for me, but not in this branch. So it's something with these changes.

Looking more, it seems like the default behaviour of autodoc (e.g. if we have our `autodoc_defaults` return `None`) is to not show `self`. So these changes might put us more in-line with what `autodoc` otherwise does. Anyway, I don't think it's a big issue, so I'll just go with it as it currently is here.